core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java

```diff
@@ -131,12 +131,13 @@ public BuildResult run() throws Exception {
                 rootBuilder.setArchiveLocation(root);
             } else {
                 rootBuilder.setArchiveLocation(effectiveModel.getAppArtifact().getPaths().iterator().next());
+
+                effectiveModel.getAppArtifact().getPaths().forEach(p -> {
```

Comment:
@aloubyansky we really need to look at how paths are passed into Quarkus bootstrap, at the moment it is super confusing. This is casing problems in the prod mode tests because it is being added from the integration tests pom.xml, so there are two roots. Because the QuarkusClassLoader creation code expects a root to still be a single file they end up missing from the ClassLoader.